HW04.Rmd

```diff
@@ -78,42 +151,114 @@ Assume that:
 ### Are whites who are killed by guns more likely to die because of suicide or homicide? How does this compare to blacks and hispanics?
 
 ```{r race}
-
+#select data and count
+race_intent <- gun_deaths %>% 
+    select(intent, race) %>% 
+    filter(intent %in% c("Suicide", "Homicide"), race %in% c("White", "Black", "Hispanic")) %>% 
+    group_by(race, intent) %>% 
+    count() %>% 
+    ungroup() %>% 
+    group_by(race) %>% 
+    mutate(total_race = sum(n)) %>% 
+    mutate(percentage = round(n / total_race * 100, digits = 2)) %>% 
+    select(-total_race)
 ```
 
 (hint maybe looking at percentages would be good)
 ```{r race_2}
-
+#plot the data and set position to "fill" to show ratio
+ggplot(race_intent, aes(x = race, y = n, fill = intent)) +
+    geom_col(position = "fill") +
+    labs(title = "Homicide and Suicide Gun Deaths by Race",
+         x = "Race",
+         y = "Ratio of Intent",
+         caption = "Source: FiveThirtyEight",
+         fill = "Intend") +
+    theme_classic() +
+    theme(plot.title = element_text(hjust = 0.5))
+knitr::kable(x = race_intent, format = "markdown", col.names = c("Race", "Intent", "Number of Deaths", "Percentage per Race"), align = "ccc", caption= "Homicide and Suicide Gun Deaths by Race")
 ```
-
+From the figure and table, whites who are killed by guns are more likely to die because of suicide, while blacks and hispanics are more likely to die because of homicide.
 
 ### Are police-involved gun deaths significantly different from other gun deaths? Assess the relationship between police involvement and age, police involvement and race, and the intersection of all three variables.
 
 ```{r police}
-
+police_age <- gun_deaths %>% 
+#select and filter data, and convert to easily manipulated type 
+  select(police, age) %>% 
+  filter(age != "Not Available") %>% 
+  mutate(police = as.factor(police), age = as.numeric(age))
+levels(police_age$police) <- c("Not Involved", "Involved")
+#plot data
+ggplot(police_age, 
+       aes(x = police, y = age, color = police)) +
+       geom_boxplot(outlier.alpha = 0) +
+       labs(title = "Relationship Between Victim Ages and Police Involvement",
+           x = "Police Involvement",
+           y = "Victim Age",
+           caption = "Source: FiveThirtyEight",
+           color = "Police") +
+       theme_classic() 
 ```
 
 Think about performing stats such as a t.test to see if differences you are plotting are statistically significant
 
 ```{r stats?}
-
+t.test(police_age$age[police_age$police == 'Not Involved'], police_age$age[police_age$police == 'Involved'])

```

Comment:
Clean way of just doing a t-test! I defined the x and y variables for the t.test(x, y) function separately. Turns out that wasn't necessary.